### PR TITLE
update balance swaap-v2

### PR DESCRIPTION
**Pull request by tien7668**
update price levels of swaap-v2

### CODE REVIEW

**Review by hanguyenk — APPROVED** [review]

File: pkg/liquidity-source/swaap-v2/pool_simulator.go [hunk 1]
```diff
@@ -204,3 +213,25 @@ func getAmountOut(amountIn float64, priceLevels []PriceLevel) (float64, error) {
...
 }
+
+func getNewPriceLevelsState(amountIn float64, priceLevels []PriceLevel) []PriceLevel {
+	if len(priceLevels) == 0 {
+		return priceLevels
+	}
+
+	for i, priceLevel := range priceLevels {
+		if amountIn > priceLevel.Level {
+			continue
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/swaap-v2/pool_simulator.go:224]: this `continue` won't go to line 231?

Comment (NgoKimPhu) [reply pkg/liquidity-source/swaap-v2/pool_simulator.go:224]: oh looks like swaap-v2 price levels are cumulative

File: pkg/liquidity-source/swaap-v2/pool_simulator.go [hunk 2]
```diff
@@ -204,3 +213,25 @@ func getAmountOut(amountIn float64, priceLevels []PriceLevel) (float64, error) {
...
+		} else if amountIn == priceLevel.Level {
+			if i == len(priceLevels)-1 {
+				return nil
+			}
+			return priceLevels[i+1:]
```
Comment (NgoKimPhu) [inline pkg/liquidity-source/swaap-v2/pool_simulator.go:229]: in other RFQ's I see they often use first price level as the minimum swap amount, but looks like swaap doesn't have this?

File: pkg/liquidity-source/swaap-v2/pool_simulator.go (hunk 1 above)
Comment (tien7668) [reply pkg/liquidity-source/swaap-v2/pool_simulator.go:224]: No, It won't. swaap-v2's price-levels are accumulated, mean at index i, the level[i] = level[i-1] + amount[i]

File: pkg/liquidity-source/swaap-v2/pool_simulator.go (hunk 2 above)
Comment (tien7668) [reply pkg/liquidity-source/swaap-v2/pool_simulator.go:229]: Yeah, haven't found any info about minimum amount from the docs and current impl. Actually, first price level we got is 0, means we can swap with any amount, I believe

**Pull request merged**